environment_scenes/World.gd

```diff
@@ -0,0 +1,19 @@
+extends Node2D
+
+
+func _process(delta):
+	
+	if $Mob.position.x < $Player.position.x:
+		$Mob/Animations.flip_h = false
+	else:
+		$Mob/Animations.flip_h = true
+		
+	if $Player/Animations.animation == "walking" || $Player/Animations.animation == "jumping":
```

Comment:
little awkward to have stats depend on what animation it's in, but meh it works

Reply:
Also it doesn't seem that the player losses stamina when jumping (is there an animation for jumping?)

Reply:
I think it's better to have some state variable instead of relying on what animation we're in. For example, if we wanted stamina to deplete faster when sprinting then we'd have to had a new animation. But with a state variable we can easily add and check.

Reply:
Wouldn't that basically be the same logic, though? Or maybe I misunderstood what you meant.

Reply:
Also, yeah, I've yet to figure out the if/elif/else stuff properly. We do enter the jumping state, but only very, very briefly (since I used if_action_just_pressed).

Reply:
It's the same logic but now there will be a dedicated state variable. The variable then can be used to manage the animations based on the state/states the player is in, along with managing stats based on if the player is in like "taken-damage" state

Reply:
Hmm, but unless I'm mistaken, the AnimatedSprite "animations" variable is already a dedicated state variable. When you update it, you're setting the sprite's current state. And then when you play() the sprite, it runs through the frames for the currently set state.

Reply:
Note: I made some commits on my end to change the stamina depletion/regen logic. I plan to do the following:

Let the Player.gd script control stamina internally, and then have World.gd regularly update its GUI/HUD to reflect these changes. In other words, Player.gd will contain "MAX_STAMINA" and "stamina" variables. When the player moves, their stamina will decrease. When they're idle, it will regenerate. The player's MAX_STAMINA will define the width of the stamina bar in the HUD. The World.gd script will then access information about the Player to:

- set the width of the stamina and health bars
- update the stamina and health bars

According to the logic defined in Player.gd

Let me know if there's a better way to go about this, though. Thanks!

Reply:
well, I guess it's fine. But if we wanted to add states that do not have any animation associated with them it'll be a bit awkward to manage them with "animations" variable

Reply:
Hmm, touche. I guess we can think on it some more for the time being.